src/Compilers/CSharp/Portable/FlowAnalysis/FlowAnalysisPass.cs

```diff
@@ -23,18 +24,41 @@ internal class FlowAnalysisPass
             BoundBlock block,
             DiagnosticBag diagnostics)
         {
+            if (method.IsScriptInitializer)
+            {
+                // For script content, we're interested in errors but
+                // not warnings (such as unused variable) that are
+                // common when compiling code snippets.
+                var allDiagnostics = DiagnosticBag.GetInstance();
+                block = Rewrite(method, block, addImplicitReturnOnly: false, diagnostics: allDiagnostics);
+                diagnostics.AddRange(allDiagnostics.AsEnumerable().Where(d => d.Severity == DiagnosticSeverity.Error));
```

Comment:
Please check with the diagnostics team. It doesn't seem right to hard-code a blanket policy in such a low layer.